### PR TITLE
New org user

### PR DESCRIPTION
**Pull request by mattkrick**
TEST
- [x] app works in node v10 & node v11 (implemented uws replacement)
- [x] a "New" badge appears news to org users that were added during the current monthly subscription cycle
- [x] Billing leader can select "Refund and Remove From Organization" for PRO new org users from the org user dropdown menu
- [x] after removing an org user from the org dropdown, the invoice estimate updates (for PRO)
- [x] users who have already been refunded & then have been put back on the org will still expire at the end of the period
- [x] calling `sendNewOrganizationUsersConfirmation` sends an email to billing leaders of organizations including the names & email addresses of new users.

### CODE REVIEW

**Comment by mattkrick** [comment]
@ackernaut email is ready for styles!
check out `UpcomingInvoiceEmail` to style it. you can test it at http://localhost:3000/email/UpcomingInvoiceEmailTemplate

File: package.json [hunk 1]
```diff
@@ -137,6 +137,7 @@
   },
   "dependencies": {
     "@babel/polyfill": "^7.0.0",
+    "@clusterws/cws": "^0.10.0",
```
Comment (mattkrick) [inline package.json]: same as uws, but kept updated by the community

File: src/server/graphql/mutations/helpers/createNewOrg.js [hunk 2]
```diff
@@ -19,7 +18,7 @@ export default async function createNewOrg (orgId, orgName, leaderUserId) {
       id: shortid.generate(),
       inactive: false,
       joinedAt: now,
-      newUserUntil: new Date(Date.now() + NEW_USER_GRACE_PERIOD),
+      newUserUntil: now,
```
Comment (mattkrick) [inline src/server/graphql/mutations/helpers/createNewOrg.js]: org creators aren't considered new

File: src/server/graphql/queries/invoiceDetails.js [hunk 3]
```diff
@@ -20,13 +20,12 @@ export default {
...
     const viewerId = getUserId(authToken)
-    const [type, maybeOrgId] = invoiceId.split('_')
-    const isUpcoming = type === 'upcoming'
+    const isUpcoming = invoiceId.startsWith('upcoming_')
     const currentInvoice = await r
       .table('Invoice')
       .get(invoiceId)
       .default(null)
-    const orgId = (currentInvoice && currentInvoice.orgId) || maybeOrgId
+    const orgId = (currentInvoice && currentInvoice.orgId) || invoiceId.substring(9) // remove 'upcoming_'
```
Comment (mattkrick) [inline src/server/graphql/queries/invoiceDetails.js]: longstanding parse bug that affects any org with an underscore in their id :see_no_evil:

File: src/server/socketHandlers/wssConnectionHandler.js [hunk 4]
```diff
@@ -12,8 +12,7 @@ import {TREBUCHET_WS} from '@mattkrick/trebuchet-client'
 
 const APP_VERSION = packageJSON.version
 export default function connectionHandler (sharedDataLoader, rateLimiter) {
-  return async function socketConnectionHandler (socket) {
-    const req = socket.upgradeReq
+  return async function socketConnectionHandler (socket, req) {
```
Comment (mattkrick) [inline src/server/socketHandlers/wssConnectionHandler.js]: websockets no longer hold onto the req. in the future, we could clear this for ~20% memory savingss

**Comment by ackernaut** [comment]
Going to take a look at this email now

File: package.json (hunk 1 above)
Comment (ackernaut) [reply package.json]: 👍

File: src/server/graphql/queries/invoiceDetails.js (hunk 3 above)
Comment (ackernaut) [reply src/server/graphql/queries/invoiceDetails.js]: doh

**Comment by ackernaut** [comment]
- [x] Read the commits
- [x] Pushed email styles/content a little farther

### Test No. 1
- [ ] app works in node v10 & node v11 (implemented uws replacement)
- [ ] a "New" badge appears news to org users that were added during the current monthly subscription cycle
- [ ] Billing leader can select "Refund and Remove From Organization" for PRO new org users from the org user dropdown menu
- [ ] after removing an org user from the org dropdown, the invoice estimate updates (for PRO)
- [ ] users who have already been refunded & then have been put back on the org will still expire at the end of the period
- [ ] calling `sendNewOrganizationUsersConfirmation` sends an email to billing leaders of organizations including the names & email addresses of new users.

**Comment by ackernaut** [comment]
So I’m seeing an error when trying to access the invoice estimate. I think my steps were:
- Remove a new user from org
- Re-invite that user to the org
- Now the invoice doesn’t show (it did before)
```
{ TypeError: Cannot read property 'amount' of undefined
    at amount (/Users/terryacker/code/parabolinc/action/src/server/billing/helpers/generateInvoice.js:119:62)
    at Array.reduce (<anonymous>)
    at reduce (/Users/terryacker/code/parabolinc/action/src/server/billing/helpers/generateInvoice.js:119:25)
    at makeQuantityChangeLineItems (/Users/terryacker/code/parabolinc/action/src/server/billing/helpers/generateInvoice.js:250:35)
  message: "Cannot read property 'amount' of undefined",
  locations: [ { line: 11, column: 3 } ],
  path: [ 'viewer', 'invoiceDetails' ] }
```
Otherwise, everything else is looking solid!!

**Comment by mattkrick** [comment]
oh this is good, lemme try to reproduce

**Comment by mattkrick** [comment]
i can't seem to reproduce.
- set invoice cache to 2 seconds
- create new org
- turn org pro
- invite new user
- accept as new user
- check invoice
- remove new user
- check in voice
- add user again
- accept again
- check invoice

anything i'm missing?

**Comment by ackernaut** [comment]
I’ll try your steps again locally.

**Comment by ackernaut** [comment]
Gonna try merging in latest `master` locally. I think the main thing to look out for are changes to `package`?

**Comment by ackernaut** [comment]
Nice work @mattkrick

**Pull request merged**